doc/plugins.dox

```diff
@@ -186,59 +195,143 @@ Let's start with the very basic headers and add more later if necessary.
 @a geanyplugin.h includes all of the Geany API and also the necessary GTK header files,
 so there is no need to include @a gtk/gtk.h yourself.
 
-@note
-@a plugindata.h contains the biggest part of the plugin API and provides some basic macros.
-
-Then you should define two basic variables which will give access to data fields
-provided by the plugin API.
-@code
-GeanyPlugin			*geany_plugin;
-GeanyData			*geany_data;
-@endcode
+@note @a plugindata.h contains the biggest part of the plugin API and provides some basic macros.
 
 Now you can go on and write your first lines for your new plugin. As mentioned before,
-you will need to implement and fill out a couple of functions/macros to make the plugin work.
-So let's start with PLUGIN_VERSION_CHECK().
-
-PLUGIN_VERSION_CHECK() is a convenient way to tell Geany which version of Geany's plugin API
-is needed at minimum to run your plugin. The value is defined in
+you will need to implement and a couple of functions. Arguably the most important
+one is @a geany_load_module(). This is the function that makes Geany see you plugin at all. When
+Geany scans the pre-defined and user-configured plugin directories, it will take a look at each
+shared library (or DLL on Windows) to see if it exports a @a geany_load_module() symbol. Files
+lacking these will be outright ignored.
+
+@subsection register Registering a Plugin
+
+Geany will then invoke this function, always and regardless of whether the user activates your
+plugin. In fact its purpose to probe if the plugin should even be presented to the user. Therefore
+it important that you use this function to register your plugin, making it available to the user.
+Geany will pass a pointer to a GeanyPlugin instance which acts as a unique handle to your plugin.
+Use this pointer for registering and later API calls. Registering the plugin consists of a number
+of steps:
+
+  1. Filling GeanyPlugin::info with metadata
```

Comment:
are these fields in the plugin instance that the pointer points to?  if so probably need to say so